numpy/ma/core.pyi

```diff
@@ -540,8 +542,12 @@ class MaskedArray(ndarray[_ShapeT_co, _DTypeT_co]):
     def filled(self, /, fill_value: _ScalarLike_co | None = None) -> ndarray[_ShapeT_co, _DTypeT_co]: ...
     def compressed(self) -> ndarray[tuple[int], _DTypeT_co]: ...
     def compress(self, condition, axis=..., out=...): ...
-    def __eq__(self, other): ...
-    def __ne__(self, other): ...
+
+    # TODO: How to deal with the non-commutative nature of `==` and `!=`?
+    # xref numpy/numpy#17368
+    def __eq__(self, other: Incomplete, /) -> Incomplete: ...
+    def __ne__(self, other: Incomplete, /) -> Incomplete: ...
```

Comment:
For now I've copied this from

https://github.com/numpy/numpy/blob/8b2321d03c1abce671a15f9d08a97249bd2d161d/numpy/__init__.pyi#L1685-L1688

I think that's a separate topic to resolve, and that this should be aligned with what's decided for `ndarray`. FWIW, I'm generally of the opinion that _some_ usual type information is better than nothing

Reply:
Yea, that's a tricky one. I agree that this is fine for now.